### PR TITLE
Deprecate ARBORX_USE_CUDA_AWARE_MPI and replace it ARBORX_USE_GPU_AWARE_NPI

### PR DESCRIPTION
**Pull request by Rombur**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -58,7 +58,16 @@ endif()
 cmake_dependent_option(ARBORX_USE_CUDA_AWARE_MPI
                        "Allow using device data in MPI communication"
                        OFF "ARBORX_ENABLE_MPI" OFF)
+if(ARBORX_USE_CUDA_AWARE_MPI)                     
+  set(ARBORX_USE_GPU_AWARE_MPI ON)
+  message(DEPRECATION "ARBORX_USE_CUDA_AWARE_MPI is deprecated use ARBORX_USE_GPU_AWARE_MPI")
```
Comment (aprokop) [inline CMakeLists.txt:63]:
```suggestion
  message(DEPRECATION "ARBORX_USE_CUDA_AWARE_MPI is deprecated, use ARBORX_USE_GPU_AWARE_MPI instead")
```

**Review by aprokop — APPROVED** [review]

**Review by dalg24 — COMMENTED** [review]
Can you provide more context about this PR?
Ideally we would detect whether the MPI implementation is GPU-aware or not, and enable both code paths so that it is possible to chose at runtime whether to explicitly copy to the host before sending/receiving

**Comment by Rombur** [comment]
> Ideally we would detect whether the MPI implementation is GPU-aware or not, and enable both code paths so that it is possible to chose at runtime whether to explicitly copy to the host before sending/receiving

Each MPI implementation has its own way to enable GPU-aware messaging. As you can see [here](https://docs.olcf.ornl.gov/systems/crusher_quick_start_guide.html#gpu-aware-mpi) on Crusher you also need to set an environment variable. I don't think that's worth it to have a bunch of code to do the detection.

I have change the name from `ARBORX_USE_GPU_AWARE_MPI` to `ARBORX_ENABLE_GPU_AWARE_MPI`. It's more in line with the way we define other options. The current message for the option already makes it clear that it **allows** us to use GPU-aware enable not that we will use when the option is enabled.

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

File: CMakeLists.txt [hunk 2]
```diff
@@ -55,10 +55,19 @@ if(ARBORX_ENABLE_MPI)
   find_package(MPI REQUIRED)
   target_link_libraries(ArborX INTERFACE MPI::MPI_CXX)
 endif()
-cmake_dependent_option(ARBORX_USE_CUDA_AWARE_MPI
+cmake_dependent_option(ARBORX_ENABLE_CUDA_AWARE_MPI
+                       "Allow using device data in MPI communication"
+                       OFF "ARBORX_ENABLE_MPI" OFF)
+if(ARBORX_USE_CUDA_AWARE_MPI)                     
+  set(ARBORX_ENABLE_GPU_AWARE_MPI ON)
+  message(DEPRECATION "ARBORX_USE_CUDA_AWARE_MPI is deprecated use ARBORX_ENABLE_GPU_AWARE_MPI")
```
Comment (aprokop) [inline CMakeLists.txt:63]:
```suggestion
    message(DEPRECATION "ARBORX_USE_CUDA_AWARE_MPI is deprecated, use ARBORX_ENABLE_GPU_AWARE_MPI instead")
```

**Review by aprokop — APPROVED** [review]

**Pull request merged**